qiskit/pulse/builder.py

```diff
@@ -1648,7 +1688,11 @@ def frequency_offset(
     finally:
         if compensate_phase:
             duration = builder.get_context().duration - t0
-            dt = active_backend().configuration().dt
+            # backendV2
+            if isinstance(active_backend(), BackendV2):
+                dt = active_backend().dt
+            else:
+                dt = active_backend().configuration().dt
```

Comment:
I see this logic to get dt everywhere in this file. This is of course not the scope of this PR, but could you please add new builder command to get backend dt in 0.25 for cleanup?

Reply:
Do you want me to add like below?

```python
def get_dt_from_backend(backend):
  if isinstance(backend. backendV2):
    return backend.dt
  else:
    return backend.configuration().dt
```

Reply:
Yes, in the followup PR for 0.25. I think this makes some logic cleaner.